src/provider/states/starting.rs

```diff
@@ -56,22 +56,43 @@ impl State<PodState> for Starting {
                     "Starting command: {:?} with arguments {:?}",
                     binary, os_args
                 );
+
+                // Check if environment variables are set on the container - if some are present
+                // we render all values as templates to replace configroot, packageroot and logroot
+                // directories in case they are referenced in the values
+                //
+                // If even one of these renderings fails the entire pod will be failed and
+                // transitioned to a complete state with the error that occurred.
+                // If all renderings work, the vec<(String,String)> is returned as value and used
+                // later when starting the process
                 let env_variables = if let Some(vars) = container.env() {
                     debug!(
                         "Got environment vars: {:?} service {}",
                         vars, pod_state.service_name
                     );
-                    vars.iter()
+                    let render_result = vars
```

Comment:
Would you mind just adding a simple comment about which "magic" is being used here? Collect to a Result of ...preferably with a link to Rustdoc